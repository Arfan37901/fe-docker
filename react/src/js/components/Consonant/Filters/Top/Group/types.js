import {
    func,
    shape,
    string,
    number,
    arrayOf,
} from 'prop-types';

import { FilterItemType } from '../../../types/config';

// eslint-disable-next-line import/prefer-default-export
export const GroupType = {
    id: string.isRequired,
    name: string.isRequired,
    clearFilterText: string,
    numItemsSelected: number,
    onCheck: func.isRequired,
    results: number.isRequired,
    onClearAll: func.isRequired,
    items: arrayOf(shape(FilterItemType)).isRequired,
};
