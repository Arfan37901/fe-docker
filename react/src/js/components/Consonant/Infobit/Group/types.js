import {
    shape,
    arrayOf,
    oneOfType,
} from 'prop-types';

import {
    FooterLeftType,
    FooterRightType,
    FooterCenterType,
} from '../../types/card';

// eslint-disable-next-line import/prefer-default-export
export const GroupType = {
    renderList: arrayOf(oneOfType([
        shape(FooterLeftType),
        shape(FooterRightType),
        shape(FooterCenterType),
    ])),
};
