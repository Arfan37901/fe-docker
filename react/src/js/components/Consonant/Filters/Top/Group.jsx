import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Footer } from './Footer';
import { Items } from './Items';

import { stopPropagation } from '../../Helpers/general';

import {
    useConfig,
    useExpandable,
} from '../../Helpers/hooks';

/**
 * Minimum quantity of the top filter options to apply blur on options' wrapper
 * @type {Number}
 */
const clipWrapperItemsCount = 9;

/**
 * Top filter
 *
 * @component
 * @example
 * const props= {
    name: String,
    id: String,
    items: Array,
    numItemsSelected: Number,
    onCheck: Function,
    onClearAll: Function,
    results: Number,
    clearFilterText: String,
 * }
 * return (
 *   <Group {...props}/>
 * )
 */
const Group = (props) => {
    const {
        name,
        id,
        items,
        numItemsSelected,
        onCheck,
        onClearAll,
        results,
        clearFilterText,
    } = props;

    const getConfig = useConfig();

    /**
     **** Authored Configs ****
     */
    const mobileGroupTotalResultsText =
        getConfig('filterPanel', 'i18n.topPanel.mobile.group.totalResultsText').replace('{total}', results);
    const mobileGroupApplyBtnText = getConfig('filterPanel', 'i18n.topPanel.mobile.group.applyBtnText');
    const mobileGroupDoneBtnText = getConfig('filterPanel', 'i18n.topPanel.mobile.group.doneBtnText');

    /**
     **** Hooks ****
     */

    /**
     * @typedef {String} openDropdownState -  Id of the top filter
     * @description â€” defined in the component props
     *
     * @typedef {Function} ToggleSetter - Handles toggling opened/closed state of
     * the top filter
     * @description
     *
     * @type {[String, Function]} openDropdown
     */
    const [openDropdown, handleToggle] = useExpandable(id);

    /**
     **** Constants ****
     */

    /**
     * Whether the top filter is opened
     * @type {Boolean}
     */
    const isOpened = openDropdown === id;

    /**
     * Whether at least one top filter option is selected
     * @type {Boolean}
     */
    const atleastOneItemSelected = numItemsSelected > 0;

    /**
     * Text of the top filter footer button:
     * whether the "Apply changes" text should be shown or "Done"
     * @type {String}
     */
    const mobileFooterBtnText =
        atleastOneItemSelected ? mobileGroupApplyBtnText : mobileGroupDoneBtnText;

    /**
     * Handles unselection of the top filter options
     * when the "Clear filter options" link is clicked
     * @param {ClickEvent} e
     * @listens ClickEvent
     */
    const handleClear = (e) => {
        e.stopPropagation();
        onClearAll(id);
    };

    /**
     * Handles toggling the selected/unselected state of the top filter option
     * when the filter option is checked/unchecked
     * @param {ChangeEvent} e
     * @listens ChangeEvent
     */
    const handleCheck = (e) => {
        e.stopPropagation();
        onCheck(id, e.target.value, e.target.checked);
    };

    /**
     * Array of the top filter selected options
     * @type {Array}
     */
    const selectedFilters = items.filter(item => item.selected);

    /**
     * Whether at least one top filter option is selected
     * @type {Boolean}
     */
    const atleastOneFilterSelected = selectedFilters.length > 0;

    /**
     * Text indicating the quantity of the selected top filter options
     * @type {String}
     */
    const selectedItemQtyText = selectedFilters.length > 0 ? `${selectedFilters.length}` : '';

    /**
     * Whether the top filter is opened
     * @type {Boolean}
     */
    const filterGroupOpened = isOpened;

    /**
     * Whether the top filter is closed
     * @type {Boolean}
     */
    const filterGroupNotOpened = !isOpened;

    /**
     * Whether the top filter options should be blurred at the bottom of the parent container
     * @type {Boolean}
     */
    const shouldClipFilters = items.length >= clipWrapperItemsCount;

    /**
     * Class name for the top filter:
     * whether the top filter is opened
     * whether the top filter contains at least one selected option
     * @type {String}
     */
    const containerClassname = classNames({
        'consonant-top-filter': true,
        'consonant-top-filter_opened': filterGroupOpened,
        'consonant-top-filter_selected': atleastOneFilterSelected && filterGroupNotOpened,
    });

    return (
        <div
            data-testid="filter-item"
            className={containerClassname}>
            <div
                className="consonant-top-filter--inner">
                <h3
                    className="consonant-top-filter--name">
                    <button
                        type="button"
                        className="consonant-top-filter--link"
                        data-testid="filter-group-btn"
                        onClick={handleToggle}
                        tabIndex="0">
                        {name}
                        <span
                            className="consonant-top-filter--selcted-items-qty">
                            {selectedItemQtyText}
                        </span>
                    </button>
                </h3>
                <div
                    className="consonant-top-filter--selcted-items">
                    <div
                        className="consonant-top-filter--absolute-wrapper">
                        {<Items
                            clipWrapperItemsCount={clipWrapperItemsCount}
                            handleCheck={handleCheck}
                            stopPropagation={stopPropagation}
                            items={items} />
                        }
                        {shouldClipFilters &&
                            <aside
                                className="consonant-top-filter--bg" />
                        }
                        {<Footer
                            mobileFooterBtnText={mobileFooterBtnText}
                            handleToggle={handleToggle}
                            clearFilterText={clearFilterText}
                            handleClear={handleClear}
                            numItemsSelected={numItemsSelected}
                            mobileGroupTotalResultsText={mobileGroupTotalResultsText} />
                        }
                    </div>
                </div>
            </div>
        </div>
    );
};

Group.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onCheck: PropTypes.func.isRequired,
    onClearAll: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    numItemsSelected: PropTypes.number,
    results: PropTypes.number.isRequired,
    clearFilterText: PropTypes.string,
};

Group.defaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
};

/* eslint-disable-next-line import/prefer-default-export */
export { Group };
