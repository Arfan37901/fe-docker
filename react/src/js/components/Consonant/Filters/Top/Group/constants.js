export const defaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
};

/**
 * Minimum quantity of the top filter options to apply blur on options' wrapper
 * @type {Number}
 */
export const clipWrapperItemsCount = 9;
