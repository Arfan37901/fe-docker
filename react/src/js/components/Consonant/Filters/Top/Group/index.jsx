import React, {
    useMemo,
    useCallback,
} from 'react';
import classNames from 'classnames';

import { Items } from '../Items';
import { Footer } from '../Footer';
import { selector } from './utils';
import { GroupType } from './types';
import { If } from '../../../Common';
import {
    defaultProps,
    clipWrapperItemsCount,
} from './constants';
import {
    template,
    stopPropagation,
} from '../../../Helpers/general';
import {
    useExpandable,
    useConfigSelector,
} from '../../../Helpers/hooks';


/**
 * Top filter
 *
 * @component
 * @example
 * const props= {
    name: String,
    id: String,
    items: Array,
    numItemsSelected: Number,
    onCheck: Function,
    onClearAll: Function,
    results: Number,
    clearFilterText: String,
 * }
 * return (
 *   <Group {...props}/>
 * )
 */
const Group = ({
    id,
    name,
    items,
    onCheck,
    results,
    onClearAll,
    clearFilterText,
    numItemsSelected,
}) => {
    /**
     **** Authored Configs ****
     */
    const {
        doneButtonText,
        applyButtonText,
        totalResultTextTemplate,
    } = useConfigSelector(selector);

    const totalResultText = template(totalResultTextTemplate, { total: results });

    /**
     **** Hooks ****
     */

    /**
     * @typedef {String} openDropdownState -  Id of the top filter
     * @description â€” defined in the component props
     *
     * @typedef {Function} ToggleSetter - Handles toggling opened/closed state of
     * the top filter
     * @description
     *
     * @type {[String, Function]} openDropdown
     */
    const [openDropdown, handleToggle] = useExpandable(id);

    /**
     **** Constants ****
     */

    /**
     * Whether the top filter is opened
     * @type {Boolean}
     */
    const isOpened = openDropdown === id;

    /**
     * Whether at least one top filter option is selected
     * @type {Boolean}
     */
    const atleastOneItemSelected = numItemsSelected > 0;

    /**
     * Text of the top filter footer button:
     * whether the "Apply changes" text should be shown or "Done"
     * @type {String}
     */
    const mobileFooterBtnText =
        atleastOneItemSelected ? applyButtonText : doneButtonText;

    /**
     * Handles unselection of the top filter options
     * when the "Clear filter options" link is clicked
     * @param {ClickEvent} e
     * @listens ClickEvent
     */
    const handleClear = useCallback((event) => {
        event.stopPropagation();
        onClearAll(id);
    }, [id, onClearAll]);

    /**
     * Handles toggling the selected/unselected state of the top filter option
     * when the filter option is checked/unchecked
     * @param {ChangeEvent} e
     * @listens ChangeEvent
     */
    const handleCheck = useCallback((event) => {
        const { target: { value, checked } } = event;

        event.stopPropagation();

        onCheck(id, value, checked);
    }, [id, onCheck]);

    /**
     * Array of the top filter selected options
     * @type {Array}
     */
    const selectedFilters = useMemo(() => items.filter(item => item.selected), [items]);

    /**
     * Whether at least one top filter option is selected
     * @type {Boolean}
     */
    const atleastOneFilterSelected = selectedFilters.length > 0;

    /**
     * Text indicating the quantity of the selected top filter options
     * @type {String}
     */
    const selectedItemQtyText = selectedFilters.length > 0 ? `${selectedFilters.length}` : '';

    /**
     * Whether the top filter is opened
     * @type {Boolean}
     */
    const filterGroupOpened = isOpened;

    /**
     * Whether the top filter is closed
     * @type {Boolean}
     */
    const filterGroupNotOpened = !isOpened;

    /**
     * Whether the top filter options should be blurred at the bottom of the parent container
     * @type {Boolean}
     */
    const shouldClipFilters = items.length >= clipWrapperItemsCount;

    /**
     * Class name for the top filter:
     * whether the top filter is opened
     * whether the top filter contains at least one selected option
     * @type {String}
     */
    const containerClassname = classNames({
        'consonant-top-filter': true,
        'consonant-top-filter_opened': filterGroupOpened,
        'consonant-top-filter_selected': atleastOneFilterSelected && filterGroupNotOpened,
    });

    return (
        <div
            data-testid="filter-item"
            className={containerClassname}>
            <div
                className="consonant-top-filter--inner">
                <h3
                    className="consonant-top-filter--name">
                    <button
                        tabIndex="0"
                        type="button"
                        onClick={handleToggle}
                        data-testid="filter-group-btn"
                        className="consonant-top-filter--link">
                        {name}
                        <span
                            className="consonant-top-filter--selcted-items-qty">
                            {selectedItemQtyText}
                        </span>
                    </button>
                </h3>
                <div
                    className="consonant-top-filter--selcted-items">
                    <div
                        className="consonant-top-filter--absolute-wrapper">
                        <Items
                            items={items}
                            handleCheck={handleCheck}
                            stopPropagation={stopPropagation}
                            clipWrapperItemsCount={clipWrapperItemsCount} />
                        <If condition={Boolean(shouldClipFilters)}>
                            <aside
                                className="consonant-top-filter--bg" />
                        </If>
                        <Footer
                            handleClear={handleClear}
                            handleToggle={handleToggle}
                            clearFilterText={clearFilterText}
                            numItemsSelected={numItemsSelected}
                            mobileFooterBtnText={mobileFooterBtnText}
                            mobileGroupTotalResultsText={totalResultText} />
                    </div>
                </div>
            </div>
        </div>
    );
};

Group.propTypes = GroupType;
Group.defaultProps = defaultProps;

/* eslint-disable-next-line import/prefer-default-export */
export { Group };
