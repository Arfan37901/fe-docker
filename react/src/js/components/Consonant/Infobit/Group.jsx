import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Bookmark from './Type/Bookmark';
import Button from './Type/Button';
import Icon from './Type/Icon';
import IconWithText from './Type/IconWithText';
import TextLink from './Type/Link';
import LinkWithIcon from './Type/LinkWithIcon';
import Price from './Type/Price';
import Progress from './Type/Progress';
import Rating from './Type/Rating';
import Text from './Type/Text';

const INFOBIT_TYPE = {
    PRICE: 'price',
    BUTTON: 'button',
    ICON_TEXT: 'icon-with-text',
    LINK_ICON: 'link-with-icon',
    TEXT: 'text',
    ICON: 'icon',
    LINK: 'link',
    PROGRESS: 'progress-bar',
    RATING: 'rating',
    BOOKMARK: 'bookmark',
};

function Group(props) {
    const { renderList } = props;
    const renderItems = (data) => {
        let arr = data;
        const res = [];

        if (!data) throw new Error();
        if (data && !Array.isArray(data) && typeof data === 'object') arr = [data];

        try {
            arr.forEach((el) => {
                switch (el.type) {
                    case INFOBIT_TYPE.PRICE:
                        res.push(<Price />);
                        break;
                    case INFOBIT_TYPE.BUTTON:
                        res.push(<Button {...el} />);
                        break;
                    case INFOBIT_TYPE.ICON_TEXT:
                        res.push(<IconWithText />);
                        break;
                    case INFOBIT_TYPE.LINK_ICON:
                        res.push(<LinkWithIcon />);
                        break;
                    case INFOBIT_TYPE.TEXT:
                        res.push(<Text />);
                        break;
                    case INFOBIT_TYPE.ICON:
                        res.push(<Icon />);
                        break;
                    case INFOBIT_TYPE.LINK:
                        res.push(<TextLink />);
                        break;
                    case INFOBIT_TYPE.PROGRESS:
                        res.push(<Progress />);
                        break;
                    case INFOBIT_TYPE.RATING:
                        res.push(<Rating />);
                        break;
                    case INFOBIT_TYPE.BOOKMARK:
                        res.push(<Bookmark />);
                        break;
                    default: break;
                }
            });

            return res;
        } catch (e) {
            alert(1);
            return res;
        }
    };

    return (<Fragment>{renderItems(renderList)}</Fragment>);
}

Group.propTypes = {
    renderList: PropTypes.arrayOf(PropTypes.shape({
        type: PropTypes.string,
    })),
};

Group.defaultProps = {
    renderList: [],
};

export default Group;
