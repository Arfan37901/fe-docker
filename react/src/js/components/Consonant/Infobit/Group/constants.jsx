import { INFOBIT_TYPE } from '../../Helpers/constants';
import {
    Icon,
    Text,
    Link,
    Price,
    Button,
    Rating,
    Progress,
    Bookmark,
    IconWithText,
    LinkWithIcon,
    DateInterval,
} from '../Type';

export const defaultProps = {
    renderList: [],
};

export const COMPONENTS = {
    [INFOBIT_TYPE.TEXT]: Text,
    [INFOBIT_TYPE.ICON]: Icon,
    [INFOBIT_TYPE.LINK]: Link,
    [INFOBIT_TYPE.PRICE]: Price,
    [INFOBIT_TYPE.BUTTON]: Button,
    [INFOBIT_TYPE.RATING]: Rating,
    [INFOBIT_TYPE.DATE]: DateInterval,
    [INFOBIT_TYPE.BOOKMARK]: Bookmark,
    [INFOBIT_TYPE.PROGRESS]: Progress,
    [INFOBIT_TYPE.ICON_TEXT]: IconWithText,
    [INFOBIT_TYPE.LINK_ICON]: LinkWithIcon,
    [INFOBIT_TYPE.LINK_ICON]: LinkWithIcon,
};
