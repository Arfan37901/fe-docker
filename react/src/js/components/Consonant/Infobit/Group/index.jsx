import React from 'react';
import uuid from 'react-uuid';

import { GroupType } from './types';
import { defaultProps, COMPONENTS } from './constants';

/**
 * Group of Infobits (shown in 3:2 Card Footer)
 *
 * @component
 * @example
 * const props= {
    renderList: Array,
 * }
 * return (
 *   <Group {...props}/>
 * )
 */
const Group = ({ renderList }) => renderList.map(({ type, ...infobit }) => {
    const Component = COMPONENTS[type];

    return !Component ? null : <Component key={uuid()} {...infobit} />;
});

Group.propTypes = GroupType;
Group.defaultProps = defaultProps;

export default Group;
