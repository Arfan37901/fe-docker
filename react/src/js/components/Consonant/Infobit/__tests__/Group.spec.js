import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import mockconfig from '../../Helpers/Testing/Mocks/consonant.json';
import ContextProvider from '../../Helpers/Testing/Utils/ContextProvider';
import Group from '../Group';

const ProvidedComponent = props => (
    <ContextProvider context={mockconfig}>
        <Group {...props} />
    </ContextProvider>
);

const GROUPS = [
    { type: 'price', price: 'price', term: 'term' },
    { type: 'button' },
    { type: 'icon-with-text' },
    { type: 'link-with-icon', href: '/some/href/', text: 'text' },
    { type: 'text', text: 'text' },
    { type: 'icon', src: 'icon' },
    { type: 'link', href: '/some/href/', text: 'text' },
    { type: 'progress-bar' },
    {
        type: 'rating',
        label: 'label',
        totalStars: 5,
        startsFilled: 3,
    },
    {
        type: 'bookmark',
        cardId: '1',
        onClick: jest.fn,
        disableBookmarkIco: true,
    },
    {
        type: 'date-interval',
        startTime: '',
        endTime: '',
        locale: '',
        dateFormat: '{LLL} {dd} | {timeRange} {timeZone}',
    },
];

describe('Consonant/Bookmarks', () => {
    test('should correct render', async () => {
        const tree = renderer
            .create(<ProvidedComponent renderList={GROUPS} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('should correct render not array', async () => {
        const tree = renderer
            .create(<ProvidedComponent renderList={{ type: 'price', price: 'price', term: 'term' }} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
