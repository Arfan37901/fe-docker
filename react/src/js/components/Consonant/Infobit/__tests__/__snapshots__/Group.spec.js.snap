// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Bookmarks should correct render 1`] = `
Array [
  <span
    className="consonant-price-infobit"
  >
    <strong
      className="consonant-price-infobit--price"
    >
      price
    </strong>
    <span
      className="consonant-price-infobit--term"
    >
      term
    </span>
  </span>,
  <a
    className="consonant-btn-infobit consonant-btn-infobit_cta"
    data-testid="consonant-btn-infobit"
    href=""
    rel="noreferrer"
    tabIndex="0"
    target="_blank"
  >
    
  </a>,
  <div
    className="consonant-icon-with-text-infobit"
  >
    
    <span
      className="consonant-icon-with-text-infobit--text"
    >
      
    </span>
  </div>,
  <a
    className="consonant-link-widh-ico-infobit"
    href="/some/href/"
    rel="noreferrer"
    tabIndex="0"
    target="__blank"
    title=""
  >
    
    <span>
      text
    </span>
  </a>,
  <p
    className="consonant-text-infobit"
  >
    text
  </p>,
  <img
    alt=""
    className="consonant-icon-infobit"
    loading="lazy"
    src="icon"
    width="28"
  />,
  <a
    className="consonant-link-infobit"
    href="/some/href/"
    rel="noreferrer"
    tabIndex="0"
    target="_blank"
    title=""
  >
    text
  </a>,
  <div
    className="consonant-progress-infobit"
  >
    <div
      className="consonant-progress-infobit--wrapper"
    >
      <span
        className="consonant-progress-infobit--text"
        title=""
      >
        
      </span>
      <span
        className="consonant-progress-infobit--text consonant-progress-infobit--text_italic"
        title=""
      >
        
      </span>
    </div>
    <div
      className="consonant-progress-infobit--el"
    >
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow={0}
        className="consonant-progress-infobit--val"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#1473E6",
            "width": "calc(0 + 2px)",
          }
        }
      >
        Progress: 
        0
      </span>
    </div>
  </div>,
  <div
    className="consonant-rating-infobit"
    data-stars={5}
  >
    <span
      className="consonant-rating-infobit--stars"
      data-rating={0}
    />
    <span
      className="consonant-rating-infobit--text"
    >
      label
    </span>
  </div>,
  <button
    className="consonant-bookmark-infobit consonant-bookmark-infobit_disabled"
    data-testid="bookmark-button"
    data-tooltip-wrapper={true}
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="consonant-bookmark-infobit--ico"
      data-testid="bookmarks--ico"
      style={
        Object {
          "backgroundImage": "",
        }
      }
    />
    <span
      className="consonant-tooltip"
    >
      Save Card
    </span>
  </button>,
  <span
    className="consonant-date-interval-infobit"
    title="Invalid DateTime Invalid DateTime | Invalid Interval Invalid DateTime"
  >
    Invalid DateTime Invalid DateTime | Invalid Interval Invalid DateTime
  </span>,
]
`;

exports[`Consonant/Bookmarks should correct render not array 1`] = `
<span
  className="consonant-price-infobit"
>
  <strong
    className="consonant-price-infobit--price"
  >
    price
  </strong>
  <span
    className="consonant-price-infobit--term"
  >
    term
  </span>
</span>
`;
